code/modules/power/generator.dm

```diff
@@ -140,17 +150,30 @@
 	var/datum/browser/popup = new(user, "teg", "Thermo-Electric Generator", 460, 300)
 	popup.set_content(get_menu())
 	popup.open()
+*/
 
-/obj/machinery/power/generator/Topic(href, href_list)
-	if(..())
-		return
-	if( href_list["close"] )
-		usr << browse(null, "window=teg")
-		usr.unset_machine()
-		return FALSE
-	return TRUE
+/obj/machinery/power/generator/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ThermalGenerator", name)
+		ui.open()
+		ui.set_autoupdate(TRUE)
 
+/obj/machinery/power/generator/ui_data(mob/user)
+//	var/datum/gas_mixture/cold_circ_air1 = cold_circ.airs[1] //outlet cold
+	var/datum/gas_mixture/cold_circ_air2 = cold_circ.airs[2] //inlet cold
+//	var/datum/gas_mixture/hot_circ_air1 = hot_circ.airs[1] //outlet hot
+	var/datum/gas_mixture/hot_circ_air2 = hot_circ.airs[2] //inlet hot
```

Comment:
Any reason for the commented out vars?

Reply:
For future work. They each hold the pressures and temperatures of *before* and *after* for each circulator, but for now we are just sending the *before* data.